src/test/resources/examples/singleAllOf/controllers/ktor/Controllers.kt

```diff
@@ -66,3 +71,29 @@ public interface TestController {
             BadRequestException("Header " + name + " is required")
     }
 }
+
+/**
+ * Decorator for Ktor's ApplicationCall that provides type safe variants of the [respond] functions.
+ *
+ * It can be used as a drop-in replacement for [io.ktor.server.application.ApplicationCall].
+ *
+ * @param R The type of the response body
+ */
+public class TypedApplicationCall<R : Any> private constructor(
+    private val applicationCall: ApplicationCall,
+) : ApplicationCall by applicationCall {
+    @Suppress("unused")
+    public suspend inline fun <reified T : R> respondTyped(message: T) {
+        respond(message)
+    }
+
+    @Suppress("unused")
+    public suspend inline fun <reified T : R> respondTyped(status: HttpStatusCode, message: T) {
+        respond(status, message)
+    }
+
+    public companion object {
+        public fun <R : Any> from(applicationCall: ApplicationCall): TypedApplicationCall<R> =
+            TypedApplicationCall<R>(applicationCall)
+    }
```

Comment:
I like the general idea. I haven't looked at the implementation yet or thought about potential pitfalls. For the moment I'm mostly wondering: why a private constructor with a factory function if the factory function is the exact same as the constructor, just with a different name?

Reply:
The factory function is indeed not needed 👍 I have removed it.